File: Plugins/Aspose_Barcode_Java_for_Ruby/lib/asposebarcodejava/BarcodeImage/savebarcodeimagetostreams.rb
module Asposebarcodejava
  module SaveBarcodeImageToTtreams
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate barcode object
        builder = Rjb::import('com.aspose.barcode.BarCodeBuilder').new

        # Set symbology of the barcode
        builder.setSymbologyType(Rjb::import('com.aspose.barcode.Symbology').Code128)

        # Set code text
        builder.setCodeText("123456")

        # Initialize ByteArrayOutputStream object
        out_stream = Rjb::import('java.io.ByteArrayOutputStream').new

        # Save barcode image
        builder.save(out_stream, Rjb::import('com.aspose.barcode.BarCodeImageFormat').Jpeg)

        # Display Status.
        puts "Save Barcode Image to Streams, please check the output file."
    end
  end
end

File: Examples/src/main/java/com/aspose/barcode/examples/barcode_image/utility_features/SaveBarcodeImageToStreams.java
package com.aspose.barcode.examples.barcode_image.utility_features;

import java.io.ByteArrayOutputStream;
import java.io.IOException;

import com.aspose.barcode.generation.BarcodeGenerator;
import com.aspose.barcode.BarCodeImageFormat;
import com.aspose.barcode.EncodeTypes;

public class SaveBarcodeImageToStreams {

	public static void main(String[] args) throws IOException {
		// ExStart: SaveBarcodeImageToStreams
		BarcodeGenerator generator = new BarcodeGenerator(EncodeTypes.CODE_128, "123456");
		ByteArrayOutputStream outStream = new ByteArrayOutputStream();
		generator.save(outStream, BarCodeImageFormat.JPEG);
		// ExEnd: SaveBarcodeImageToStreams
	}
}
